src/utils/lock.js

```diff
@@ -39,10 +39,10 @@ module.exports = class Lock {
       }
 
       this[PRIVATE.WAITING].add(tryToAcquire)
-      timeoutId = setTimeout(
-        () => reject(new KafkaJSLockTimeout(this[PRIVATE.TIMEOUT_ERROR_MESSAGE]())),
-        this[PRIVATE.TIMEOUT]
-      )
+      timeoutId = setTimeout(() => {
+        reject(new KafkaJSLockTimeout(this[PRIVATE.TIMEOUT_ERROR_MESSAGE]()))
```

Comment:
Does order matter here? When `reject` is called, will any `catch` be invoked synchronously or will it happen in a subsequent tick? I'm thinking if we need to delete the item from the waiting list before rejecting, in case someone tries to reacquire the lock directly in a catch?

Reply:
As far as I understand the order doesn't matter. 

But, that said: I also would feel more "comfortable" with having the reject as the last statement -- but that adds one more line that also makes it look a bit more ugly:
```js
      timeoutId = setTimeout(() => {
        const error = new KafkaJSLockTimeout(this[PRIVATE.TIMEOUT_ERROR_MESSAGE]()) 
        this[PRIVATE.WAITING].delete(tryToAcquire)
        reject(error)
      }, this[PRIVATE.TIMEOUT])
```

The reason here is that I didn't want to change the count in the error message -- right now it shows the number of waiters _including the one failing right now_. If you change that to show the waiters _left after this one failed_:
```js
      timeoutId = setTimeout(() => {
        this[PRIVATE.WAITING].delete(tryToAcquire)
        reject(new KafkaJSLockTimeout(this[PRIVATE.TIMEOUT_ERROR_MESSAGE]()))
      }, this[PRIVATE.TIMEOUT])
```

(Or of course just bump the count in the error message helper function, but that then would require a comment -- which is a bit "meh")

Reply:
I think the first option is fine. It is a bit unintuitive since there's nothing to indicate that the error will contain a reference to the waiting list, but it's fine.